### ISSUE TITLE
Errors involving overload resolution should show the inferred type of the argument

### ISSUE THREAD

**Issue by amluto** [astral-sh/ty#2716]
### Summary

I wrote some code that (correctly) failed type-checking.  The code reduced to:

```python
import ctypes
import weakref

map: weakref.WeakValueDictionary[int, object] = weakref.WeakValueDictionary()

value = ctypes.c_void_p(5).value
map.get(value, None)
```

ty says:

```
error[no-matching-overload]: No overload of bound method `get` matches arguments
 --> ty_bad_error.py:7:1
  |
6 | value = ctypes.c_void_p(5).value
7 | map.get(value, None)
  | ^^^^^^^^^^^^^^^^^^^^
  |
info: First overload defined here
   --> stdlib/weakref.pyi:126:9
    |
124 |     def __deepcopy__(self, memo: Any) -> Self: ...
125 |     @overload
126 |     def get(self, key: _KT, default: None = None) -> _VT | None: ...
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
127 |     @overload
128 |     def get(self, key: _KT, default: _VT) -> _VT: ...
    |
info: Possible overloads for bound method `get`:
info:   (self, key: _KT@WeakValueDictionary, default: None = None) -> _VT@WeakValueDictionary | None
info:   (self, key: _KT@WeakValueDictionary, default: _VT@WeakValueDictionary) -> _VT@WeakValueDictionary
info:   [_T](self, key: _KT@WeakValueDictionary, default: _T) -> _VT@WeakValueDictionary | _T
info: rule `no-matching-overload` is enabled by default

Found 1 diagnostic
```

If `ty` is going to tell me that no overload matches, IMO it would be nice if it told me what arguments it thinks I tried to pass.  (In this case, the issue is that `ctypes.c_void_p.value` returns None if the value is numerically 0, which I think is ridiculous and certainly not obvious.  `ty` knows this and could have told me that the type I passed was `int | None`).

I tried to include a playground link but the SHARE button had no effect.  And this issue was not assisted in the slightest by AI, except insofar as GPT 5.2 Codex reviewed the code in question and did not spot the bug :)

### Version

ty 0.0.14 (I also tried 0.0.11 with similar results)

**Comment by sharkdp** [comment]
Thank you for reporting this.

I agree that we should do this. We already do so for calls to non-overloaded functions.

**Comment by dhruvmanila** [comment]
Another idea that I thought of doing was to provide reasoning as to why the overloads got filtered out (https://github.com/astral-sh/ruff/blob/bbaa7698227fcf4f7ec001af6aa5110d405ee54f/crates/ty_python_semantic/src/types/call/bind.rs#L4328-L4330), specifically the ones that are easier to implement and is understandable from a user perspective.
